tests/unit/test_graphstates.py

```diff
@@ -111,8 +111,37 @@ def test_SCZ_mat(self, random_graph):
             assert np.array_equal(mat[N:, :N], random_graph[1])
             assert np.array_equal(mat[N:, N:], np.identity(N))
 
-    # def test_SCZ_apply(self):
-    # pass
+    @pytest.mark.parametrize("one_shot", [True, False])
+    @pytest.mark.parametrize("n", [1, 2])
+    def test_SCZ_apply(self, random_graph, one_shot, n):
+        """Test SCZ matrix application."""
+
+        adj = random_graph[1]
+        SCZ = SCZ_mat(adj)
+        N = adj.shape[0]
+        quads = np.ones([N * 2] * n)
+
+        if n == 1:
+            expected_quads = SCZ_mat(adj).dot(quads)
+        else:
+            expected_quads = SCZ.dot(SCZ.dot(quads).T).T
+
+        new_quads = SCZ_apply(adj, quads, one_shot=one_shot)
+
+        assert np.allclose(new_quads, expected_quads)
+
+    @pytest.mark.parametrize("one_shot", [True, False])
+    def test_SCZ_apply_twodimensional(self, random_graph, one_shot):
+        """Test SCZ matrix application to two-dimensional arrays."""
+
+        adj = random_graph[1]
+        N = adj.shape[0]
+        quads = np.ones((N * 2, N * 2))
+
+        new_quads = SCZ_apply(adj, quads, one_shot=one_shot)
+        SCZ = SCZ_mat(adj)
+        expected_quads = SCZ.dot(SCZ.dot(quads).T).T
+        assert np.allclose(new_quads, expected_quads)
```

Comment:
I think this is already checked in the n = 2 case of the previous test - is that correct?